library/general/noupdateserviceaccount/template.yaml

```diff
@@ -0,0 +1,99 @@
+apiVersion: templates.gatekeeper.sh/v1beta1
+kind: ConstraintTemplate
+metadata:
+  name: noupdateserviceaccount
+  annotations:
+    description: "Blocks updating the service account on resources that abstract over Pods"
+spec:
+  crd:
+    spec:
+      names:
+        kind: NoUpdateServiceAccount
+      validation:
+        openAPIV3Schema:
+          properties:
+            allowedGroups:
+              description: Groups that should be allowed to bypass the policy.
+              type: array
+              items: string
+            allowedUsers:
+              description: Users that should be allowed to bypass the policy.
+              type: array
+              items: string
+  targets:
+  - target: admission.k8s.gatekeeper.sh
+    rego: |
+      package noupdateserviceaccount
+
+      privileged(userInfo, allowedUsers, allowedGroups) {
+        # Allow if the user is in allowedUsers.
+        # Use object.get so omitted parameters can't cause policy bypass by
+        # evaluating to undefined.
+        username := object.get(userInfo, "username", "")
```

Comment:
Since this rego looks for `username` from `userInfo`, it will definitely fail when in audit, but since it also only applies to `UPDATE` operations, which is also not available in audit, it's essentially a pass thru policy. We should document this policy does NOT work for audit.

Reply:
That's a really good point - it gives me a question unrelated to the PR so I'll ask in gatekeeper-maintainers.

Reply:
Where is the best place to document that? Just in the docstrings?

Reply:
we can add in annotation and readme in the `noupdateserviceaccount` folder perhaps

Reply:
Thanks, added.